mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -342,7 +359,14 @@ def event(self, request, id):
                 status=404,
             )
 
-        provider = models.Provider.objects.get(pk=provider_id)
+        if device.provider_id != data_broker_id:
+            msg = "%s is trying to push an event with the provider id %s" % (
+                data_broker_id,
+                device.provider_id,
+            )
+            logger.warning(msg)
```

Comment:
Same question on the warning, but don't you want to prevent it with some ValidationError if the the user is just an operator?